packages/odie-client/src/data/use-get-support-interactions.ts

```diff
@@ -17,9 +17,20 @@ export const useGetSupportInteractions = (
 ) => {
 	const path = `?per_page=${ per_page }&page=${ page }&status=${ status }`;
 	const isTestMode = isTestModeEnvironment();
+	const { data: canConnectToZendesk } = useCanConnectToZendeskMessaging( enabled );
+	// Only fetch Zendesk interactions if the user can connect to Zendesk.
+	const shouldFetch = enabled && ( canConnectToZendesk || ! provider?.includes( 'zendesk' ) );
```

Comment:
As per discussion above, this can lead to assume that the code can only fetch Zendesk interactions, if the code is not read very carefully.

I'd slightly change it to make it clearer (albeit more verbose):
```suggestion
	// Only fetch interactions if the user can connect to the provider.
	let shouldFetch = enabled;
	if ( provider?.includes( 'zendesk' ) && ! canConnectToZendesk ) {
		shouldFetch = false;
	}
```

Reply:
This is perfect. Thanks!